libs/dyn/mapping.go

```diff
@@ -41,7 +41,8 @@ func newMappingWithSize(size int) Mapping {
 func newMappingFromGoMap(vin map[string]Value) Mapping {
 	m := newMappingWithSize(len(vin))
 	for k, v := range vin {
-		m.Set(V(k), v)
+		// QQQ log error or panic?
+		_ = m.Set(V(k), v)
```

Comment:
This can be a panic. Set returns an error only if the specified key is not a string.

Reply:
If Set requires keys to be string, why not accept string?

`func Set(key string, value Value) /* no error possible */`

Reply:
The key is a `dyn.Value` because it has location information associated with it.

We can add a separate `SetWithString(key string, value Value)` to cover the cases where the input does _not_ have location information associated with it. This is the case when we iterate over a `map[string]Value`, for example, but not when we iterate over a map that originates from YAML.

Reply:
To be safe, I've reverted all changes to libs/dyn and annotated relevant lines with //nolint:errcheck.

That way we can land this PR safely and improve libs/dyn at a later time.